packages/sonarwhal/src/lib/utils/resource-loader.ts

```diff
@@ -232,7 +232,7 @@ export const loadResource = (name: string, type: ResourceType, configurations: A
         throw new Error(`Resource ${name} isn't compatible with current sonarwhal version`);
     }
 
-    if (!resource) {
+    if (!resource || !resource.name || !equalNormalizedString(resource.name, `${name}${type}`)) {
```

Comment:
I think `resource.name` is not going to exists. To read the name you need to read the `package.json` of the package (you can use something like `require(path.join(process.cwd(), 'package.json'))`

Reply:
@sarvaje Hmm...I don't see `name` in package.json either, should that parameter be added to package.json first? Also I tried loading connectors/rules/parsers/external rules and `resource.name` is always the same as the exported class name in each package.

Reply:
take a look to any package in the folder `packages` and take a look to their `package.json` you will see the `name` property.

Reply:
> you can use something like `require(path.join(process.cwd(), 'package.json'))`

You should be using [`getPackage()` from `misc`](https://github.com/sonarwhal/sonarwhal/blob/cec7d5bed384cddae5bf23a2d54005b00562c2b7/packages/sonarwhal/src/lib/utils/misc.ts#L312), that way it will be easier to mock if needed.

Reply:
@sarvaje For the package that contains multiple rules (such as `jsll`), the `name` property in package.json is `@sonarwhal/packagename` (@sonarwhal/jsll), it doesn't contain the name of the rule to be loaded. I can verify the package name, but to do that it will require that we always use `packageName/rule-ruleName` in `.sonarwhal` when there are multiple rules in a package. Will this be okay?

Reply:
> but to do that it will require that we always use `packageName/rule-ruleName` in `.sonarwhal` when there are multiple rules in a package

This is what we should be doing already